File: src/main/java/codewars/DiffOf2Collections.java
package codewars;

import java.util.Collection;
import java.util.List;
import java.util.stream.Collectors;

public class DiffOf2Collections {
    public static List<Character> diff(Collection<Character> a, Collection<Character> b) {

        List<Character> list = a.stream()
                .filter(item -> !b.contains(item))
                .collect(Collectors.toList());

        return list;
    }
}

File: src/main/kotlin/codewarskt/DiffOf2Collections.kt
package codewarskt


class DiffOf2Collections {

    fun diff(listA: Collection<Char>, listB: Collection<Char>): List<Char> {
        val setA = listA.toSet().filter { !listB.contains(it) }
        val setB = listB.toSet().filter { !listA.contains(it) }
        val resultSet = setA + setB
        return resultSet.toList()
    }
}